modules/gapi/include/opencv2/gapi/streaming/onevpl/default.hpp

```diff
@@ -0,0 +1,56 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2022 Intel Corporation
+
+#ifndef OPENCV_GAPI_STREAMING_ONEVPL_UTILS_HPP
+#define OPENCV_GAPI_STREAMING_ONEVPL_UTILS_HPP
+
+#include <opencv2/gapi/streaming/source.hpp>
+#include <opencv2/gapi/streaming/onevpl/cfg_params.hpp>
+#include <opencv2/gapi/streaming/onevpl/data_provider_interface.hpp>
+#include <opencv2/gapi/streaming/onevpl/device_selector_interface.hpp>
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+enum class VPLSourcePlatform : int {
+    CPU,
+    GPU
+};
+
+/**
+ * @brief G-API streaming source based on OneVPL with a default configuration.
+ *
+ * The default configuration assumes using ID3D11Device,
+ * ID3D11DeviceContext, mfxImplDescription.AccelerationMode and MFX_ACCEL_MODE_VIA_D3D11.
+ *
+ * @note stream sources are passed to G-API via shared pointers, so
+ *  please use gapi::make_onevpl_src<> to create objects and ptr() to pass a
+ *  GSource to cv::gin().
+ * 
+ * @note device and context pointers can be utilized for inference while
+ * frames produced by source can be processed with other backends (e.g. OCL).
+ */
+class GAPI_EXPORTS DefaultVPLSource {
```

Comment:
how do we explain to users or how we may describe the necessity of using **TWO**  VPL sources in documentation ?
It's a confusing moment.

 So DefaultVPLSource looks useless and redundant

Reply:
I'm renaming it to something else to avoid the confusion. We need a source plus 2 extra pointers connected to that source for remote context